### PR TITLE
l10n: fix various interpolations

### PR DESCRIPTION
**Pull request by dwaynebailey**
Seems we have a few strings that are doing the interpolation in a non-robust way and also in a way that ensures that they can't be localised.

* ``_('A %s b' % var)`` - is broken as it can't be localised
* ``_('A %s b') % var`` - is more correct
* ``_('A %s b', var)`` - is, I believe, more robust

### CODE REVIEW

File: pootle/apps/pootle_project/models.py [hunk 1]
```diff
@@ -388,7 +388,7 @@ def delete(self, *args, **kwargs):
     def clean(self):
         if self.code in RESERVED_PROJECT_CODES:
             raise ValidationError(
-                _('"%s" cannot be used as a project code' % (self.code,))
+                _('"%s" cannot be used as a project code') % (self.code,)
```
Comment (unho) [inline pootle/apps/pootle_project/models.py]: Why not `_('"%s" cannot be used as a project code', self.code)`?

File: pootle/apps/import_export/utils.py [hunk 2]
```diff
@@ -27,26 +27,26 @@ def import_file(file, user=None):
...
+                                      "X-Pootle-Revision header\n",
+                                      file.name))
     rev = int(rev)
 
     try:
         store, created = Store.objects.get_or_create(pootle_path=pootle_path)
     except Exception as e:
         raise FileImportError(_("Could not create '%s'. Missing "
-                                "Project/Language? (%s)") % (file.name, e))
+                                "Project/Language? (%s)", file.name, e))
```
Comment (unho) [inline pootle/apps/import_export/utils.py]: Perhaps the tuple should be kept.

File: pootle/apps/pootle_project/models.py (hunk 1 above)
Comment (dwaynebailey) [inline pootle/apps/pootle_project/models.py]: To be honest really wasn't sure about that one

File: pootle/apps/import_export/utils.py (hunk 2 above)
Comment (dwaynebailey) [inline pootle/apps/import_export/utils.py]: I don't now TBH

Comment (dwaynebailey) [inline pootle/apps/import_export/utils.py]: Seems you're right, broke the build

File: pootle/apps/pootle_project/models.py (hunk 1 above)
Comment (unho) [inline pootle/apps/pootle_project/models.py]: Well, you changed it in other places to be like this.

Comment (dwaynebailey) [inline pootle/apps/pootle_project/models.py]: Except its a (x,) instance.

Comment (unho) [inline pootle/apps/pootle_project/models.py]: No, it doesn't. Using `_('"%s" cannot be used as a project code', self.code)` should just work.

**Comment by dwaynebailey** [comment]
@translate/core looking for a GTM here please

**Comment by jleclanche** [comment]
lgtm.

BTW, I noticed you turned `% (conn_settings)` into `% conn_settings`. As a rule of thumb, I always parenthesize string arguments to string formats. It's a pass-through syntax wise and has two benefits:
* When adding a new argument, the syntax is immediately clear (no need to add parentheses)
* It distinguishes string arguments to tuple or dynamic arguments. Eg the case where you do

```py
args = "foo", "bar", "baz"
s = "hi %s, hello %s, greetings %s" % args
```

File: pootle/checks.py [hunk 3]
```diff
@@ -114,8 +114,8 @@ def check_redis(app_configs=None, **kwargs):
         workers = Worker.all(queue.connection)
     except Exception as e:
         conn_settings = queue.connection.connection_pool.connection_kwargs
-        errors.append(checks.Critical(_("Could not connect to Redis (%s)") % (e),
-            hint=_("Make sure Redis is running on %(host)s:%(port)s") % (conn_settings),
+        errors.append(checks.Critical(_("Could not connect to Redis (%s)", e),
+            hint=_("Make sure Redis is running on %(host)s:%(port)s") % conn_settings,
```
Comment (unho) [inline pootle/checks.py]: Just curious why the `%` is used since in other places you switched to a comma.

Comment (jleclanche) [inline pootle/checks.py]: It's correct. ugettext doesn't support dict formatting. cf
https://docs.djangoproject.com/en/1.8/topics/i18n/translation/#standard-translation

**Comment by dwaynebailey** [comment]
@jleclanche thanks for the review.  My logic for ``conn_settings`` was that it is a dictionary, so the brackets where redundant and actually a bit confusing.   I might have misunderstood your comment though.  So I'm landing without it unless you can help with my broken understanding.

**Comment by jleclanche** [comment]
As I mentioned on IRC, I was actually agreeing with your change - feel free to land it. Sorry it didn't quite come through ;)

**Pull request merged**